File: backend/app/schemas/report.py
from datetime import date, datetime
from typing import Optional
from pydantic import BaseModel, Field


class ReportUploadResponse(BaseModel):
    id: int
    jenis_jasa: str
    nama_file: str
    review_status: str
    uploaded_at: Optional[datetime] = None


class ReportUpdateRequest(BaseModel):
    jenis_jasa: str
    tanggal_permintaan: Optional[date] = None
    tenaga_produksi: Optional[str] = None
    tanggal_laporan: Optional[date] = None
    nomor_laporan: Optional[str] = None
    debitur: Optional[str] = None
    master_bank_id: Optional[int] = None
    perbankan: Optional[str] = None
    perihal_pekerjaan: Optional[str] = None
    lokasi_objek: Optional[str] = None
    resi_pengiriman: Optional[str] = None
    blok: Optional[str] = None
    no_urut: Optional[str] = None
    notes: Optional[str] = None
    save_mode: str = Field(pattern="^(draft|final)$")


class ReportDetailResponse(BaseModel):
    id: int
    jenis_jasa: str
    tanggal_permintaan: Optional[date] = None
    tenaga_produksi: Optional[str] = None
    tanggal_laporan: Optional[date] = None
    nomor_laporan: Optional[str] = None
    debitur: Optional[str] = None
    master_bank_id: Optional[int] = None
    perbankan: Optional[str] = None
    perihal_pekerjaan: Optional[str] = None
    lokasi_objek: Optional[str] = None
    resi_pengiriman: Optional[str] = None
    blok: Optional[str] = None
    no_urut: Optional[str] = None
    review_status: str
    nama_file: str
    file_url: Optional[str] = None
    uploaded_at: Optional[datetime] = None
    reviewed_at: Optional[datetime] = None
    notes: Optional[str] = None

    class Config:
        from_attributes = True

File: frontend/types/report.ts
export type ApiResponse<T> = {
  success: boolean;
  message: string;
  data: T;
};

export type ReportListItem = {
  id: number;
  jenis_jasa: string;
  tanggal_laporan: string | null;
  nomor_laporan: string | null;
  debitur: string | null;
  perbankan: string | null;
  review_status: string;
  nama_file: string;
  uploaded_at: string | null;
};

export type ReportDetail = {
  id: number;
  jenis_jasa: string;
  tanggal_permintaan: string | null;
  tenaga_produksi: string | null;
  tanggal_laporan: string | null;
  nomor_laporan: string | null;
  debitur: string | null;
  master_bank_id: number | null;
  perbankan: string | null;
  perihal_pekerjaan: string | null;
  lokasi_objek: string | null;
  resi_pengiriman: string | null;
  blok: string | null;
  no_urut: string | null;
  review_status: string;
  nama_file: string;
  file_url: string | null;
  uploaded_at: string | null;
  reviewed_at: string | null;
  notes: string | null;
};

export type ReportTypeOption = {
  code: string;
  label: string;
};